tests/manage/pv_services/space_reclaim/test_rbd_reclaimspace_cronjob.py

```diff
@@ -0,0 +1,138 @@
+        )
+
+    @polarion_id("OCS-2759")
+    @tier1
+    def test_rbd_space_reclaim_cronjob(self):
+        """
+        Test to verify RBD space reclamation
+        Steps:
+        1. Create and attach RBD PVC of size 25 GiB to an app pod.
+        2. Get the used size of the RBD pool
+        3. Create four files of size 4GiB
+        4. Verify the increased used size of the RBD pool
+        5. Delete three file
+        6. Create ReclaimSpaceJob
+        7. Verify the decreased used size of the RBD pool
+        8. Verify the presence of other files in the folder
+        """
+        pvc_obj = self.pvc[0]
+        pod_obj = self.pod[0]
+
+        fio_filename1 = "fio_file1"
+        fio_filename2 = "fio_file2"
+        fio_filename3 = "fio_file3"
+        fio_filename4 = "fio_file4"
+
+        schedule = ["hourly", "midnight", "weekly"]
+        # Fetch the used size of pool
+        cbp_name = self.sc_obj.get().get("parameters").get("pool")
+        log.info("Cephblock pool name {cbp_name}")
+        used_size_before_io = fetch_used_size(cbp_name)
+        log.info(f"Used size before IO is {used_size_before_io}")
+
+        # Create three 4 GB file
+        for filename in [fio_filename1, fio_filename2, fio_filename3, fio_filename4]:
+            pod_obj.run_io(
+                storage_type="fs",
+                size="4G",
+                runtime=100,
+                fio_filename=filename,
+                end_fsync=1,
+            )
+            pod_obj.get_fio_results()
+
+        # Verify used size after IO
+        exp_used_size_after_io = used_size_before_io + (16 * self.pool_replica)
+        used_size_after_io = fetch_used_size(cbp_name, exp_used_size_after_io)
+        log.info(f"Used size after IO is {used_size_after_io}")
+
+        # Delete the file and validate the reclaimspace cronjob
+        for filename in [fio_filename1, fio_filename2, fio_filename3]:
+            file_path = get_file_path(pod_obj, filename)
+            pod_obj.exec_cmd_on_pod(
+                command=f"rm -f {file_path}", out_yaml_format=False, timeout=100
+            )
+
+            # Verify file is deleted
+            try:
+                check_file_existence(pod_obj=pod_obj, file_path=file_path)
+            except CommandFailed as cmdfail:
+                if "No such file or directory" not in str(cmdfail):
+                    raise
+                log.info(f"Verified: File {file_path} deleted.")
+
+            # Create ReclaimSpaceCronJob
+            for type in schedule:
+                reclaim_space_job = pvc_obj.create_reclaim_space_cronjob(schedule=type)
+
+            # Wait for the Succeeded result of ReclaimSpaceJob
+            try:
+                for reclaim_space_job_yaml in TimeoutSampler(
+                    timeout=120, sleep=5, func=reclaim_space_job.get
+                ):
+                    result = reclaim_space_job_yaml["spec"]["schedule"]
+                    if result == "@" + type:
```

Comment:
Isn't this verifying only the job having schedule "weekly" which is the  last item in the list 'schedule' ?

Reply:
> Isn't this verifying only the job having schedule "weekly" which is the last item in the list 'schedule' ?

This has to be under the for loop. will change it.